clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -42,192 +77,142 @@
-    private ApplicationEventHandler applicationEventHandler;
-    private SubscriptionState subscriptions;
+    private AsyncKafkaConsumer<String, String> consumer = null;
 
-    @BeforeEach
-    public void setup() {
-        // By default, the consumer is part of a group and autoCommit is enabled.
-        setup(ConsumerTestBuilder.createDefaultGroupInformation(), true);
-    }
-
-    private void setup(Optional<ConsumerTestBuilder.GroupInformation> groupInfo, boolean enableAutoCommit) {
-        testBuilder = new ConsumerTestBuilder.AsyncKafkaConsumerTestBuilder(groupInfo, enableAutoCommit, true);
-        applicationEventHandler = testBuilder.applicationEventHandler;
-        consumer = testBuilder.consumer;
-        fetchCollector = testBuilder.fetchCollector;
-        subscriptions = testBuilder.subscriptions;
-    }
+    private final Time time = new MockTime(1);
+    private final FetchCollector<String, String> fetchCollector = mock(FetchCollector.class);
+    private final ApplicationEventHandler applicationEventHandler = mock(ApplicationEventHandler.class);
+    private final ConsumerMetadata metadata = mock(ConsumerMetadata.class);
+    private final LinkedBlockingQueue<BackgroundEvent> backgroundEventQueue = new LinkedBlockingQueue<>();
 
     @AfterEach
-    public void cleanup() {
-        if (testBuilder != null) {
-            shutDown();
+    public void resetAll() {
+        backgroundEventQueue.clear();
+        if (consumer != null) {
+            consumer.close();
         }
+        consumer = null;
+        Mockito.framework().clearInlineMocks();
     }
 
-    private void shutDown() {
-        prepAutocommitOnClose();
-        testBuilder.close();
+    private AsyncKafkaConsumer<String, String> setup() {
+        final Properties props = requiredConsumerProperties();
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, "group-id");
+        final ConsumerConfig config = new ConsumerConfig(props);
+        return setup(config);
     }
 
-    private void resetWithEmptyGroupId() {
-        // Create a consumer that is not configured as part of a group.
-        cleanup();
-        setup(Optional.empty(), false);
+    private AsyncKafkaConsumer<String, String> setupWithoutGroupId() {
+        final Properties props = requiredConsumerProperties();
+        final ConsumerConfig config = new ConsumerConfig(props);
+        return setup(config);
     }
 
-    private void resetWithAutoCommitEnabled() {
-        cleanup();
-        setup(ConsumerTestBuilder.createDefaultGroupInformation(), true);
+    @SuppressWarnings("UnusedReturnValue")
+    private AsyncKafkaConsumer<String, String> setupWithEmptyGroupId() {
+        final Properties props = requiredConsumerPropertiesAndGroupId("");
+        final ConsumerConfig config = new ConsumerConfig(props);
+        return setup(config);
+    }
+
+    private AsyncKafkaConsumer<String, String> setup(ConsumerConfig config) {
```

Comment:
nit: 
Setup is quite generic. What about:
```suggestion
    private AsyncKafkaConsumer<String, String> newConsumer(ConsumerConfig config) {
```

Reply:
Done